datamap/bloom/src/main/java/org/apache/carbondata/datamap/bloom/AbstractBloomDataMapWriter.java

```diff
@@ -173,7 +173,13 @@ protected void addValue2BloomIndex(int indexColIdx, Object value) {
     } else {
       if (indexColumns.get(indexColIdx).hasEncoding(Encoding.DICTIONARY)
           || indexColumns.get(indexColIdx).hasEncoding(Encoding.DIRECT_DICTIONARY)) {
-        indexValue = convertDictionaryValue(indexColIdx, (byte[]) value);
+        if (value.getClass().isArray()) {
```

Comment:
Maybe we can add a variable to indicate whether it is from 'Rebuild(builder0' or 'DirectWrite(writer)'. Use the class type as condition is not proper.
Do not forget to add comments for that variable.